src/hex_filename.erl

```diff
@@ -0,0 +1,55 @@
+% Excerpt from https://github.com/erlang/otp/blob/OTP-20.0.1/lib/stdlib/src/filename.erl#L761-L788
+% with modifications for changing local function calls to remote function calls
+% to the `filename` module, for the functions `pathtype/1`, `split/1`, and `join/1`
```

Comment:
Why is that required?

Reply:
Because we only vendor one function instead of the whole file.

Reply:
I thought you could just use the stdlib functions without vendoring? What prevents that?

Reply:
I've added a rationale to that file which is that `safe_relative_path` was not present in previous OTP releases.